MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs

```diff
@@ -817,895 +329,84 @@ private string GetComparableName(string name)
-
-            var seriesXmlFilename = language.ToLowerInvariant() + ".xml";
-
-            return Path.Combine(seriesDataPath, seriesXmlFilename);
-        }
-
-        /// <summary>
-        /// Gets the series data path.
-        /// </summary>
-        /// <param name="appPaths">The app paths.</param>
-        /// <returns>System.String.</returns>
-        internal static string GetSeriesDataPath(IApplicationPaths appPaths)
-        {
-            var dataPath = Path.Combine(appPaths.CachePath, "tvdb");
-
-            return dataPath;
-        }
-
-        private void DeleteXmlFiles(string path)
-        {
-            try
-            {
-                foreach (var file in _fileSystem.GetFilePaths(path, true)
-                    .ToList())
+                catch (TvDbServerException e)
                 {
-                    _fileSystem.DeleteFile(file);
+                    _logger.LogError(e, "Failed to find series end date for series {TvdbId}", tvdbSeries.Id);
                 }
             }
-            catch (IOException)
-            {
-                // No biggie
-            }
         }
 
-        /// <summary>
-        /// Sanitizes the XML file.
-        /// </summary>
-        /// <param name="file">The file.</param>
-        /// <returns>Task.</returns>
-        private async Task SanitizeXmlFile(string file)
+        private static void MapActorsToResult(MetadataResult<Series> result, IEnumerable<Actor> actors)
         {
-            string validXml;
-
-            using (var fileStream = _fileSystem.GetFileStream(file, FileOpenMode.Open, FileAccessMode.Read, FileShareMode.Read, true))
-            {
-                using (var reader = new StreamReader(fileStream))
-                {
-                    var xml = await reader.ReadToEndAsync().ConfigureAwait(false);
-
-                    validXml = StripInvalidXmlCharacters(xml);
-                }
-            }
-
-            using (var fileStream = _fileSystem.GetFileStream(file, FileOpenMode.Create, FileAccessMode.Write, FileShareMode.Read, true))
+            foreach (Actor actor in actors)
             {
-                using (var writer = new StreamWriter(fileStream))
+                var personInfo = new PersonInfo
                 {
-                    await writer.WriteAsync(validXml).ConfigureAwait(false);
-                }
-            }
-        }
-
-        /// <summary>
-        /// Strips the invalid XML characters.
-        /// </summary>
-        /// <param name="inString">The in string.</param>
-        /// <returns>System.String.</returns>
-        public static string StripInvalidXmlCharacters(string inString)
-        {
-            if (inString == null) return null;
-
-            var sbOutput = new StringBuilder();
-            char ch;
+                    Type = PersonType.Actor,
+                    Name = (actor.Name ?? string.Empty).Trim(),
+                    Role = actor.Role,
+                    ImageUrl = actor.Image,
```

Comment:
and this url has no prefix... something fishy is going on

Reply:
Forgot